workspace/__lib__/xod/graphics/text-colored/patch.cpp

```diff
@@ -0,0 +1,59 @@
+
+// clang-format off
+{{#global}}
+#include <Text.h>
+{{/global}}
+// clang-format on
+
+struct State {
+    uint8_t mem[sizeof(Text)];
+    Text* textColored;
+
+    char* str;
+    size_t cap;
+
+    uint8_t textSize;
+    int16_t x, y;
+    XColor textColor;
+};
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+
+    auto state = getState(ctx);
+    auto gfx = getValue<input_GFX>(ctx);
+
+    if (isSettingUp()) {
+        state->cap = getValue<input_L>(ctx);
+        state->str = new char[state->cap + 1];
+        state->textColored = new (state->mem) Text(gfx, state->str, state->cap);
+    }
+
+    auto str = getValue<input_S>(ctx);
+    memset(state->str, '\0', state->cap + 1);
+    dump(str, state->str);
```

Comment:
:scream: `XString` is a memory-efficient string representation. And you ruin it here, basically, creating a big-buffer copy of everything. Why can’t you just use `XString` directly?